api/v1alpha1/istiorevisiontags_types.go

```diff
@@ -0,0 +1,199 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	IstioRevisionTagKind = "IstioRevisionTag"
+	DefaultRevisionTag   = "default"
+)
+
+// IstioRevisionTagSpec defines the desired state of IstioRevisionTag
+type IstioRevisionTagSpec struct {
+	// +kubebuilder:validation:Required
+	TargetRef IstioRevisionTagTargetReference `json:"targetRef"`
+}
+
+// IstioRevisionTagTargetReference can reference either Istio or IstioRevision objects in the cluster.
+type IstioRevisionTagTargetReference struct {
+	// Kind is the kind of the target resource.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+
+	// Name is the name of the target resource.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
```

Comment:
Should this be `required` along with `Kind`?

Reply:
done! thanks